### PR TITLE
IITC-Mobile: fix some bugs in plugins handling

### LINKED ISSUE #210: Plugin read/update issues
I've enconteured two problems:

1. Read issue. When in-script key-value headers are separated with tabs (instead of spaces) the app will crash and will not be able to start properly until the plugin is deleted or tabs are replaced with spaces.
This happened to me when i've updated a plugin with tabs and encountered that iitcm was crashing :/

2. Update issue. When `plugin-name.meta.js` file exists and is readable, and a request to `plugin-name.user.js` returns almost anything the file placed on `plugins` folder will be replaced with that response. That response could be a 404 html, or something.
Possible solutions:
-- Check content-type http header
-- Check for in-script version / id headers

### PR DESCRIPTION
**Pull request by HubertZhang**
- allow tab as separator in metadata block
- do not update plugin to random html returned on 404.
  (more to fix in #257)

Fix #210

### CODE REVIEW

File: mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_FileManager.java [hunk 1]
```diff
@@ -103,16 +103,18 @@ public static void copyStream(final InputStream inStream, final OutputStream out
         map.put("description", "");
         map.put("category", "Misc");
         final BufferedReader reader = new BufferedReader(new StringReader(header));
-        String headerLine;
         try {
+            String headerLine;
             while ((headerLine = reader.readLine()) != null) {
                 if (headerLine.matches("//.*@.*")) {
```
Comment (johnd0e) [inline mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_FileManager.java]:
```suggestion
                 if (headerLine.matches("^s*//\s*@.+")) {
```
Shouldn't it be like this?

File: mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_FileManager.java [hunk 2]
```diff
@@ -103,16 +103,18 @@ public static void copyStream(final InputStream inStream, final OutputStream out
...
                     // get start of key name (first @ in line)
                     final String[] keyStart = headerLine.split("@", 2);
                     // split key value
-                    final String[] keyValue = keyStart[1].split(" ", 2);
+                    final String[] keyValue = keyStart[1].split("[ \t]", 2);
```
Comment (johnd0e) [inline mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_FileManager.java]:
```suggestion
                    final String[] keyValue = keyStart[1].split("\s", 2);
```
And here?

**Comment by johnd0e** [comment]
@HubertZhang 
Our base branch is `master`.

`test-builds` is temporary — we rewrite it often.

File: mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_FileManager.java (hunk 1 above)
Comment (HubertZhang) [reply mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_FileManager.java]: Actually I would prefer Scanner here, I would send a new commit soon

**Comment by johnd0e** [comment]
@HubertZhang 
Please fix message of commit 01038b12ec0d62b066728106f4dcece0e2d63a26, it wrongly mentions #191

**Comment by MysticJay** [comment]
Android 9.0.5 (OxigenOS on Oneplus3)
build 0.9.1.2190905.142213
- Screen shot does not work (clears layers, counts down, then offers apps to share with). Creating empty map.
- move to players location does not work. no reaction at all after tap or doubletap

**Comment by MysticJay** [comment]
Recommendation: drop Screenshot, use Android's build in fetaure.
its not fullscreen anyway, limited to current frame size.

File: mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_FileManager.java [hunk 3]
```diff
@@ -103,16 +105,13 @@ public static void copyStream(final InputStream inStream, final OutputStream out
         map.put("description", "");
         map.put("category", "Misc");
         final BufferedReader reader = new BufferedReader(new StringReader(header));
-        String headerLine;
         try {
+            Pattern p = Pattern.compile("^\\s*//\\s*@(\\S+)(.*)$");
```
Comment (johnd0e) [inline mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_FileManager.java]:
```suggestion
            final Pattern p = Pattern.compile("^\\s*//\\s*@(\\S+)(.*)$");
```

File: mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_FileManager.java [hunk 4]
```diff
@@ -103,16 +105,13 @@ public static void copyStream(final InputStream inStream, final OutputStream out
...
+            String headerLine;
             while ((headerLine = reader.readLine()) != null) {
-                if (headerLine.matches("//.*@.*")) {
-                    // get start of key name (first @ in line)
-                    final String[] keyStart = headerLine.split("@", 2);
-                    // split key value
-                    final String[] keyValue = keyStart[1].split(" ", 2);
-                    // remove whitespaces from string begin and end and push to map
-                    map.put(keyValue[0].trim(), keyValue[1].trim());
+                Matcher m = p.matcher(headerLine);
```
Comment (johnd0e) [inline mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_FileManager.java]:
```suggestion
                final Matcher m = p.matcher(headerLine);
```

File: mobile/app/src/main/java/org/exarhteam/iitc_mobile/async/UpdateScript.java [hunk 5]
```diff
@@ -95,6 +104,24 @@ private boolean isUpdateAllowed(final String url) throws MalformedURLException {
         return !mForceSecureUpdates;
     }
 
+    private String downloadFile(final String url) throws IOException {
+        Response response = client.newCall(
```
Comment (johnd0e) [inline mobile/app/src/main/java/org/exarhteam/iitc_mobile/async/UpdateScript.java]:
```suggestion
        final Response response = client.newCall(
```

File: mobile/app/src/main/java/org/exarhteam/iitc_mobile/async/UpdateScript.java [hunk 6]
```diff
@@ -24,29 +19,31 @@
...
 
     @Override
     protected Boolean doInBackground(final String... urls) {
         try {
-            mFilePath = urls[0];
+            String filePath = urls[0];
             // get local script meta information
-            mScript = IITC_FileManager.readStream(new FileInputStream(new File(mFilePath)));
-            mScriptInfo = IITC_FileManager.getScriptInfo(mScript);
+            String script = IITC_FileManager.readStream(new FileInputStream(new File(filePath)));
```
Comment (johnd0e) [inline mobile/app/src/main/java/org/exarhteam/iitc_mobile/async/UpdateScript.java]:
```suggestion
            final String script = IITC_FileManager.readStream(new FileInputStream(new File(filePath)));
```

File: mobile/app/src/main/java/org/exarhteam/iitc_mobile/async/UpdateScript.java [hunk 7]
```diff
@@ -24,29 +19,31 @@
...
+            HashMap<String, String> mScriptInfo = IITC_FileManager.getScriptInfo(script);
```
Comment (johnd0e) [inline mobile/app/src/main/java/org/exarhteam/iitc_mobile/async/UpdateScript.java]:
```suggestion
            final HashMap<String, String> mScriptInfo = IITC_FileManager.getScriptInfo(script);
```

File: mobile/app/src/main/java/org/exarhteam/iitc_mobile/async/UpdateScript.java [hunk 8]
```diff
@@ -24,29 +19,31 @@
...
 
+            mScriptName = mScriptInfo.get("name");
             String updateURL = mScriptInfo.get("updateURL");
             String downloadURL = mScriptInfo.get("downloadURL");
```
Comment (johnd0e) [inline mobile/app/src/main/java/org/exarhteam/iitc_mobile/async/UpdateScript.java:48]:
```suggestion
             final String downloadURL = mScriptInfo.get("downloadURL");
```

File: mobile/app/src/main/java/org/exarhteam/iitc_mobile/async/UpdateScript.java [hunk 9]
```diff
@@ -105,7 +102,7 @@ private boolean isUpdateAllowed(final String url) throws MalformedURLException {
     }
 
     private String downloadFile(final String url) throws IOException {
-        Response response = client.newCall(
+        Response response = mClient.newCall(
```
Comment (johnd0e) [inline mobile/app/src/main/java/org/exarhteam/iitc_mobile/async/UpdateScript.java]:
```suggestion
        final Response response = mClient.newCall(
```

File: mobile/app/src/main/java/org/exarhteam/iitc_mobile/async/UpdateScript.java (hunk 8 above)
Comment (HubertZhang) [reply mobile/app/src/main/java/org/exarhteam/iitc_mobile/async/UpdateScript.java:48]: Emmm this cannot be declared final

**Pull request merged**